spring-web/src/main/java/org/springframework/web/filter/OncePerRequestFilter.java

```diff
@@ -129,10 +129,7 @@ private boolean skipDispatch(HttpServletRequest request) {
 		if (isAsyncDispatch(request) && shouldNotFilterAsyncDispatch()) {
 			return true;
 		}
-		if (request.getAttribute(WebUtils.ERROR_REQUEST_URI_ATTRIBUTE) != null && shouldNotFilterErrorDispatch()) {
-			return true;
-		}
-		return false;
+		return request.getAttribute(WebUtils.ERROR_REQUEST_URI_ATTRIBUTE) != null && shouldNotFilterErrorDispatch();
```

Comment:
Although the proposed changed is technically correct, please revert this for the sake of readability.